Documentation/Relationships.md

```diff
@@ -0,0 +1,165 @@
+
+```
+{
+  "author": {
+    "id": 6,
+    "name": "Gob Bluth"
+  },
+  "text": "I've made a huge mistake."
+}
+```
+
+Our `Post` model could then be:
+
+```swift
+struct Post {
+  let author: String
+  let text: String
+}
+```
+
+And, our implementation of `Decodable` for `Post` looks like this:
+
+```swift
+extension Post: Decodable {
+  static func decode(j: JSON) -> Decoded<Post> {
+    return curry(self.init)
+      <^> j <| ["author", "name"]
+      <*> j <| "text"
+  }
+}
+```
+
+Great! Now we can decode JSON into `Post` models. However, let's be real, we
+can't have posts without comments! Comments are like 90% of the fun on the
+internet.
+
+Most likely the JSON will contain an embedded array of `Comment` models:
+
+```
+{
+  "author": {
+    "id": 53,
+    "name": "Lindsay"
+  },
+  "text": "I have the afternoon free.",
+  "comments": [
+    {
+      "author": {
+        "id": 1,
+        "name": "Lucille"
+      },
+      "text": "Really? Did 'nothing' cancel?"
+    }
+  ]
+}
+```
+
+So then `Comment` will look like:
+
+```swift
+struct Comment {
+  let author: String
+  let text: String
+}
+
+extension Comment: Decodable {
+  static func decode(j: JSON) -> Decoded<Comment> {
+    return curry(self.init)
+      <^> j <| ["author", "name"]
+      <*> j <| "text"
+  }
+}
+```
+
+Now, we can add an array of comments to our `Post` model:
+
+```swift
+struct Post {
+  let author: String
+  let text: String
+  let comments: [Comment]
+}
+
+extension Post: Decodable {
+  static func decode(j: JSON) -> Decoded<Post> {
+    return curry(self.init)
+      <^> j <| ["author", "name"]
+      <*> j <| "text"
+      <*> j <|| "comments"
+  }
+}
+```
+
+We added `comments` as a property on our `Post` model. Then we added a line to
+decode the comments from the JSON. Notice how we use `<||` instead of `<|` with
+`comments` because it is an _Array_.
+
+Storing the name of the author with a post or comment isn't very flexible.
+What we really want to do is tie posts and comments to users. If we use the
+`User` struct from [Basic Usage]:
+
+[Basic Usage]: Basic-Usage.md
+
+```swift
+struct User {
+  let id: Int
+  let name: String
+}
+
+extension User: Decodable {
+  static func decode(j: JSON) -> Decoded<User> {
+    return curry(User.init)
+      <^> j <| "id"
+      <*> j <| "name"
+  }
+}
+```
+
+We can simply change the `author` property from `String` to `User`. No joke!
+Take a look:
+
+```swift
+struct Post {
+  let author: User
+  let text: String
+  let comments: [Comment]
+}
+
+extension Post: Decodable {
+  static func decode(j: JSON) -> Decoded<Post> {
+    return curry(self.init)
+      <^> j <| "author"
+      <*> j <| "text"
+      <*> j <|| "comments"
+  }
+}
+
+struct Comment {
+  let author: User
+  let text: String
+}
+
+extension Comment: Decodable {
+  static func decode(j: JSON) -> Decoded<Comment> {
+    return curry(self.init)
+      <^> j <| "author"
+      <*> j <| "text"
+  }
+}
+```
+
+That's it!
```

Comment:
What do you think about wrapping up here with something mirroring the intro?

> Argo's use of polymorphism through generics makes supporting custom types straightforward, since you're using the exact same plumbing that Argo itself uses internally

or something like that